scrapper.py

```diff
@@ -31,67 +42,260 @@ class Crawler:
+                self.urls.extend(newfound[:self.max_articles_per_seed])
+                if len(self.urls) > self.total_max_articles:
+                    break
+            if len(self.urls) == old:
+                print('     Something is wrong with scraping parameters')
+                break
+
+            self.urls = self.urls[:self.total_max_articles]
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.urls
+
+
+class CrawlerRecursive(Crawler):
+
+    def __init__(self, seed_urls: list, total_max_articles: int, max_articles_per_seed: int, to_wait=False):
+        super().__init__(seed_urls, total_max_articles, max_articles_per_seed)
+        self.is_waiting = to_wait
+
+    def _crawl(self, pool: list):
+        for link in pool:
+            if self.is_waiting:
+                wait(randint(0, 10))
+            try:
+                article_bs = BeautifulSoup(requests.get(URL_START + link, 'html.parser').text, 'html.parser')
+            except requests.exceptions.ConnectionError:
+                wait(10)
+                article_bs = BeautifulSoup(requests.get(URL_START + link, 'html.parser').text, 'html.parser')
+            newfound = self._extract_url(article_bs, self.urls)
+            newfound = [i for i in newfound if len(i) > 20
+                        and not any(map(lambda y: y.isupper(), i))]
+            return newfound
+
+    def find_articles(self):
+        if self.get_backedup():
+            print('backed up urls found, starting iteration')
+        if not self.urls:
+            self.urls = self._crawl(self.seed_urls)
+            with open('links/url_backup.txt', 'w', encoding='utf-8') as file:
+                file.write('\n'.join(self.urls))
+            print(f'Scraped {len(self.urls)} from seed')
+            if self.verify_proceed():
+                print('starting recursive scraping')
+                self.find_articles()
+            else:
+                print(f'recursive crawling finished with {len(self.urls)} urls.')
+        else:
+            newfound = self._crawl(self.urls)
+            if not newfound:
+                print(f'there are no unseen links found\nrecursive crawling finished with {len(self.urls)} urls.')
+            else:
+                self.urls.extend(newfound)
+                with open('links/url_backup.txt', 'a', encoding='utf-8') as file:
+                    file.write('\n'.join(newfound))
+                print(f'found {len(newfound)} new urls')
+                if self.verify_proceed():
+                    print('starting new iteration')
+                    self.find_articles()
+                else:
+                    print(f'recursive crawling finished with {len(self.urls)} urls.')
+
+    @staticmethod
+    def verify_proceed():
+        answer = input('Would you like to proceed? yes or no: ').strip()
+        return answer == 'yes'
+
+    def get_backedup(self):
```

Comment:
is it get? or read? choose better name. and also the interface (return value) is more about checks rather than logic